### PR TITLE
Document result stdlib

### PR DESCRIPTION
**Pull request by ospencer**

### CODE REVIEW

File: src/stdlib/result.md [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+If the Result is `Ok(x)`, returns `fn(x)`. Returns the provided default otherwise.
+
+### Result.**mapWithDefaultFn**
+
+```grain
+mapWithDefaultFn : (t -> u, e -> u, Result<t, e>) -> u
+```
+
+If the Result is `Ok(x)`, returns `fnOk(x)`. If the Result is `Err(y)`, returns `fn(Err(y))`.
+
```
Comment (phated) [inline src/stdlib/result.md:84]: Did you mean?
```suggestion
If the Result is `Ok(x)`, returns `fnOk(x)`. If the Result is `Err(y)`, returns `fnErr(y)`.
```

Comment (phated) [reply src/stdlib/result.md:84]: I also think we might want to mention that both functions need to return the same type. It's indicated by the type signature, but this method was a bit confusing.

**Review by phated — APPROVED** [review]
So, other than the comments, I think this is fine to ship. However, we should really discuss the use of generics in the type signature, like `t` and `e` but then using `x` and `y` when writing descriptions. The `x` and `y` values were never mentioned in anything people can see and get really confusing. I believe they are specified that way to differentiate from the variant constructors but it just feels weird.

File: src/stdlib/result.md (hunk 1 above)
Comment (ospencer) [reply src/stdlib/result.md:84]: Yeah, that's what I meant.

**Comment by ospencer** [comment]
Yeah, definitely. What I really want are argument names in type signatures, so we can say like `(x: Number, y: Number) -> Number` with `computes the sum of x and y`.

**Pull request merged**